odxtools/utils.py

```diff
@@ -1,11 +1,29 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2022 MBition GmbH
 
-from typing import Dict, Literal, Optional, Any
+from typing import Dict, Literal, Optional, Any, Union, overload
 from xml.etree import ElementTree
 
 from .odxlink import OdxDocFragment
 
+@overload
+def str_to_bool(str_val: None) -> None: ...
+
+@overload
+def str_to_bool(str_val: str) -> bool: ...
+
+def str_to_bool(str_val: Union[None, str]) -> Union[None, bool]:
```

Comment:
simply use Optional instead of union?

Reply:
yep, thanks for the catch. that is an development artifact because I had mypy complaining here at some point, so I thought that it had an issue with `Optional`. (It turned out that the issue was something completely different...)